utils/website/graphile-config/preset.md

```diff
@@ -1,49 +1,166 @@
 ---
-sidebar_position: 3
-title: "Preset"
+sidebar_position: 2
+title: Preset
 ---
 
 # Graphile Config Preset
 
-A preset bundles together a list of plugins, and options for various of the
-"scopes". You may use more than one preset at a time, and presets may also
-compose (extend) other presets. When a library is passed a list of presets it
-results in a resolved preset (a preset that has no "extends") using the
-`ResolvePresets` algorithm; broadly all the `extends` are resolved in order, the
-plugins specified are merged as a set (each plugin will only be included once)
-and the options are merged via object merging such that the options specified
-last win.
+Target Audience: all Graphile Config users ⚙️🔌📚
 
-**NOTE**: if you compose two presets (PresetA and PresetB) that both `extends`
-the same underlying preset (BASE) and apply some overrides, then the overrides
-in PresetA will be overridden by re-applying the BASE preset again. For this
-reason, presets that are expected to be combined with other presets should not
-`extends` common/shared presets, instead the end-user should be expected to add
-these presets themselves.
+A preset bundles together a list of options and plugins to configure and extend
+a library. `GraphileConfig.Preset` is the primary interface that most Graphile
+Config users will use.
 
-**NOTE**: the order of presets is significant.
+Library consumers should define their preset as the default export of a
+`graphile.config.js` (or `.ts`, `.mjs`, etc.) file.
 
-ResolvePresets(presets):
+Here's an example in JavaScript:
+
+```ts title="graphile.config.js"
+const SomePlugin = require("some-plugin");
+
+module.exports = {
+  plugins: [SomePlugin],
+  someScope: {
+    someConfigOption: 10,
+  },
+};
+```
+
+And an equivalent configuration in TypeScript:
+
+```ts title="graphile.config.ts"
+import type {} from "graphile-config";
+import SomePlugin from "some-plugin";
+
+const preset: GraphileConfig.Preset = {
+  plugins: [SomePlugin],
+  someScope: {
+    someConfigOption: 10,
+  },
+};
+
+export default preset;
+```
+
+:::note
+
+Adding the import statement tells TypeScript about the `GraphileConfig` global
+namespace. No code from the `graphile-config` library will be included in the
+output JavaScript for `graphile.config.ts` above. See the TypeScript docs for
+more info about
+[type-only imports](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html#type-only-imports-and-export).
```

Comment:
```suggestion
:::note Importing Graphile Config types is recommended

Adding the `import type {} from "graphile-config"` statement tells TypeScript
about the `GraphileConfig` global namespace; this may or may not be necessary
depending on the other imports that you have, but we generally recommend it.

No code from the `graphile-config` library will be included in the output
JavaScript for `graphile.config.ts` above; see the TypeScript docs for more on
[type-only imports](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html#type-only-imports-and-export).
```